crates/bevy_ecs/src/reflect.rs

```diff
@@ -103,10 +103,8 @@ impl<C: Component + Reflect + FromWorld> FromType<C> for ReflectComponent {
                     .entity_mut(destination_entity)
                     .insert(destination_component);
             },
-            reflect_component: |world, entity| {
-                world
-                    .get_entity(entity)?
-                    .get::<C>()
+            reflect_component: |world, entity| unsafe {
```

Comment:
This needs a safety comment.

Reply:
this unsafe is removed in my current open PR for ptrification reviews